lib/python/qmk/search.py

```diff
@@ -0,0 +1,97 @@
+"""Functions for searching through QMK keyboards and keymaps.
+"""
+import fnmatch
+import logging
+import multiprocessing
+import re
+from dotty_dict import dotty
+from milc import cli
+
+from qmk.info import keymap_json
+import qmk.keyboard
+import qmk.keymap
+
+
+def _set_log_level(level):
+    cli.acquire_lock()
+    old = cli.log_level
+    cli.log_level = level
+    cli.log.setLevel(level)
+    logging.root.setLevel(level)
+    cli.release_lock()
+    return old
+
+
+def _all_keymaps(keyboard):
+    old = _set_log_level(logging.CRITICAL)
+    keymaps = qmk.keymap.list_keymaps(keyboard)
+    _set_log_level(old)
```

Comment:
Could add something like,
```python
import contextlib

@contextlib.contextmanager
def ignore_logging():
    old = _set_log_level(logging.CRITICAL)
    yield
    _set_log_level(old)
```

then these functions can become a little safer and do....

```py
def _all_keymaps(keyboard):
    with ignore_logging():
        keymaps = qmk.keymap.list_keymaps(keyboard)
```

Reply:
TIL. Let me have a play.